apps/web/src/app/[locale]/[username]/_components/dashboard/in-progress-tab.tsx

```diff
@@ -1,39 +1,55 @@
 import {
   Table,
   TableBody,
-  TableCaption,
   TableCell,
   TableHead,
   TableHeader,
   TableRow,
 } from '@repo/ui/components/table';
+import { prisma } from '@repo/db';
+import { cache } from 'react';
+import Link from 'next/link';
+import { getRelativeTime } from '~/utils/relativeTime';
 
-const challengesList = [
-  {
-    challengeTitle: 'Extreme Challenge 1',
-    challengeStatus: 'In-Progress',
-  },
-  {
-    challengeTitle: 'Extreme Challenge 4',
-    challengeStatus: 'In-Progress',
-  },
-];
+const getInProgressSubmissions = cache(async (id: string) => {
+  return await prisma.submission.findMany({
+    where: {
+      userId: id,
+      isSuccessful: false,
+    },
+    orderBy: [
+      {
+        createdAt: 'desc',
+      },
+    ],
+    take: 25,
+    include: {
+      challenge: true,
+    },
+    distinct: 'challengeId',
+  });
+});
+
+export async function InProgressTab({ userId }: { userId: string }) {
+  const submissions = await getInProgressSubmissions(userId);
```

Comment:
I think semantically, it is better to name this variable `challenges` because we want to display a list of challenges that are **in-progress**!